platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+14. If `lhs` != `rhs`: Return False
+15. `inner` = 0
+16. For each (`ciphertext_value`, `sid`) pair in Zip(`statements.ciphertext_values`;`statements.sids`):
+
+    16.1.  `term` =`ciphertext_value` multiplied by (`x` raised to the power of `sid`)
+
+    16.2.  `inner` =`inner` + `term`
+    
+17. `lhs` = (`inner` multiplied by `x'`) add `proof.Y`
+18. `rhs` = `g` multiplied by  `proof.z_a`
+19. If `lhs` != `rhs`: Return False
+20. `inner` = 0
+21. For each (`public_key`, `sid`) pair in Zip(`statements.public_keys`;`statements.sids`):
+
+    21.1.  `term` = `public_key` multiplied by (`proof.z_r` * (`x` raised to the power of `sid`))
+
+    21.2.  `inner` =`inner` + `term`
+22. `rhs` = `inner` + (`g` multiplied by `proof.z_a`)
+23. If `lhs` != `rhs`: Return False
+24. Return True
+
+##### Generating Participant's Private Shares & Ledger Id
+Given Participant's persistent EC PrivateKey and precisely `t` number of validated messages (t=threshold)
+each participant will decrypt all `Cᵢ` to generate an aggregated value `sᵢ` that will become a  `SecretShare(sidᵢ, sᵢ)` for each `ShareId`: `sidᵢ` owned by the participant.
+
+**Note:** All participants must choose the same set of valid `TssMessages` and have a threshold number of valid messages.
+
+![img_1.svg](img_1.svg)
+
+Also, we will extract a `PublicShare` for each `ShareId`: `sidᵢ` in the directory from the list of valid messages.
+The PublicShare for share `s` is computed by evaluating each polynomial commitment in the common set of messages at `sidᵢ` and then aggregating the results.
+
+At this point, the participant executing the scheme can start signing, sharing signatures, and validating individual signatures produced by other parties in the scheme.
+
+
+
+#####  Rekey Stage
+This Section describes the rekey process executed each time some of the scheme’s parameters change, such as the number of participants or the number of shares assigned to each participant.
+The rekeying process can happen, too, if the parameters are the same as the genesis procedure.
+
+
+The rekeying process is similar to the bootstrap process, but it starts with the previous list of owned private `SecretShare`.
+The main difference with the genesis stage is that every participant generates a `TssMessage` out of each previously owned `SecretShare`.
+
+
+![img_4.svg](img_4.svg)
+
+Once finished, the list of `SecretShare`s will be updated but the previously generated aggregate public key remains the same.
+
+
+#####  Sign Stage
+After genesis or rekeying stages, the library can sign any message.
+Using each `SecretShare` owned by the participant, a message can be signed, producing a `TssSignature`
+
+Multiple signatures can be aggregated to create an aggregate `TssSignature`. An aggregate `TssSignature` can be validated against the LedgerId if
+`t` (t=threshold) valid signatures are aggregated. If the threshold is met and the signature is valid, the library will respond with true; if not, it will be false.
```

Comment:
```suggestion
Multiple signatures can be aggregated to create an aggregate `TssSignature`. An aggregate `TssSignature` can be validated against the LedgerId (public key) if
`t` (t=threshold) valid signatures are aggregated. If the threshold is met and the signature is valid, the library will respond with true; if not, it will respond with false.
```